File: secondExercise_2/src/test/java/FeeCalculatorTest.java
import org.junit.Test;
import static org.junit.Assert.assertEquals;

public class FeeCalculatorTest {

    @Test
    public void calculateFee() {
        FeeCalculator feeCalculator = new FeeCalculator();
        Movie harry = new Movie(1, "harry");
        AmountDiscountPolicy amountDiscountPolicy = new AmountDiscountPolicy(2000, new FirstSequenceDiscountCondition());

        Screen harryInMorning = new Screen(1, harry, 6, 12, 10000, amountDiscountPolicy);

        assertEquals(feeCalculator.calculateFee(harryInMorning, 2), 16000.0, 1);
    }
}

File: secondExercise/src/test/kotlin/FeeCalculatorTest.kt
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions;

class FeeCalculatorTest {
    @Test
    internal fun `calculate fee`() {
        val feeCalculator = FeeCalculator()

        val harrypoter = Movie("해리포터", "판타지")

        val moringTimeDiscountCondition = MorningTimeDiscountCondition()

        val discountConditions = listOf<DiscountCondition>(moringTimeDiscountCondition)

        val firstSequencePolicy = AmountDiscountPolicy(discountConditions)

        val harrypoterInMorning = Screen(1, harrypoter, 9, 11, 300.0, firstSequencePolicy)

        val totalFee = feeCalculator.calculate(harrypoterInMorning, 3)

        Assertions.assertEquals(totalFee, 0.0)
    }
}
